### PR TITLE
Add camel master example

### PR DESCRIPTION
**Pull request by llowinge**

### CODE REVIEW

**Comment by llowinge** [comment]
Closes https://github.com/apache/camel-quarkus/issues/3923

**Review by oscerd — APPROVED** [review]

File: cluster-leader-election/pom.xml [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+                <groupId>${camel-quarkus.platform.group-id}</groupId>
+                <artifactId>${camel-quarkus.platform.artifact-id}</artifactId>
+                <version>${camel-quarkus.platform.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
```
Comment (jamesnetherton) [inline cluster-leader-election/pom.xml:75]: If the app is deployable to a k8s cluster then I guess we should add `camel-quarkus-microprofile-health`.

File: docs/modules/ROOT/attachments/examples.json [hunk 2]
```diff
@@ -49,6 +49,11 @@
     "description": "Shows how to define Camel routes using Kotlin programming language",
     "link": "https://github.com/apache/camel-quarkus-examples/tree/main/timer-log-kotlin"
   },
+  {
+    "title": "Leader election in Kubernetes: A Camel Quarkus Master example",
+    "description": "Shows how to use camel master component.",
```
Comment (jamesnetherton) [inline docs/modules/ROOT/attachments/examples.json:54]:
```suggestion
    "description": "Shows how to use Camel master component.",
```

Comment (llowinge) [reply docs/modules/ROOT/attachments/examples.json]: Done, thank you

File: cluster-leader-election/pom.xml (hunk 1 above)
Comment (llowinge) [reply cluster-leader-election/pom.xml:75]: Done, thank you

**Pull request merged**